src/libcollections/string.rs

```diff
@@ -1881,6 +1881,30 @@ impl Into<Vec<u8>> for String {
     }
 }
 
+#[stable(feature = "stringfromchars", since = "1.12.0")]
+impl<'a> From<&'a [char]> for String {
+    #[inline]
+    fn from(v: &'a [char]) -> String {
+        let mut s = String::with_capacity(v.len() * mem::size_of::<char>());
+        for c in v {
+            s.push(*c);
+        }
+        s
+    }
+}
+
+#[stable(feature = "stringfromchars", since = "1.12.0")]
+impl From<Vec<char>> for String {
+    #[inline]
+    fn from(v: Vec<char>) -> String {
+        let mut s = String::with_capacity(v.len() * mem::size_of::<char>());
+        for c in v {
+            s.push(c);
+        }
+        s
+    }
```

Comment:
It probably makes sense to delegate to the `&[char]` impl here instead of duplicating the code.

It's possible we could do something clever to reuse the `Vec`s buffer, though I don't think it's worth thinking about in this PR.